org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/XMLTextDocumentService.java

```diff
@@ -262,8 +263,7 @@ private XMLFormattingOptions getFormattingSettings(String uri) {
 		return computeAsync((cancelChecker) -> {
 			String uri = params.getTextDocument().getUri();
 			TextDocument document = getDocument(uri);
-			return getXMLLanguageService().format(document, null,
-					XMLFormattingOptions.create(params.getOptions(), getFormattingSettings(uri)));
+			return getXMLLanguageService().format(document, null, getSharedSettings());

```

Comment:
Since format now accepts `SharedSettings` instead of just `FormattingOptions`, for the time being, I didn't merge in the `params.getOptions` in, because the `params.getOptions` only applies to the current formatting request.

Maybe I could:
1. Merge the options in, and "unmerge" them before returning the formatting results

or

2. Make a copy of the current `SharedSettings` object, and modify that instead

Reply:
option 2 sounds better, you might want to create a new CompositeSettings(SharedSettings, FormattingOptions) to make it cleaner

Reply:
with CompositeSettings extends SharedSettings.

Reply:
Ok thank you, I have created CompositeSettings in my latest push
 https://github.com/eclipse/lemminx/blob/b7c9bb458bbeb2b99b2ccb11880ba804728212ed/org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/settings/CompositeSettings.java